testapp.c

```diff
@@ -0,0 +1,1232 @@
+    SDL_Log("F1   : help");
+    SDL_Log("F2   : save current rendering to .bmp");
+}
+
+#define HAVE_LCD 
+#define HAVE_SDF 
+
+#define HAVE_SET_FONT_SIZE_FUNCTION
+// Need patch to set size dynamically
+// https://bugzilla.libsdl.org/show_bug.cgi?id=2487
+
+#define HAVE_ALL_WRAPPED_FUNCTIONS
+// Need patch to have all wrapped functions
+// https://bugzilla.libsdl.org/show_bug.cgi?id=4361
+
+#define HAVE_WRAP_ALIGN 
+
+
+
+static int rand_n(int n);
+static void init_rand(void);
+
+static const char *test_fonts[] = {
+    "fonts/NotoColorEmoji.ttf",  // Colored Emoji
+    "fonts/FletcherGothicFLF.ttf",
+#if 1
+    /* CFF/OpenType font (.otf) */
+    "fonts/ITCAvantGardeStd-Demi.ttf", // BUG 4266
+#endif
+    "fonts/DroidSansFallback.ttf", // xstart negative Jap
+    "fonts/Папка/Lazy.ttf",
+    "fonts/DroidSans.ttf", // no kerning here ?
+    "fonts/DejaVuSans.ttf",
+    "fonts/digital-7.ttf", // BUG 969
+    "fonts/HelveticaNeue-Regular.ttf",// BUG 2622
+    "fonts/ttf_amiga/amiga4ever pro2.ttf",
+    "fonts/ttf-bitstream-vera-1.10/VeraMoBI.ttf",
+    "fonts/ttf-bitstream-vera-1.10/VeraBd.ttf",
+    "fonts/ttf-bitstream-vera-1.10/Vera.ttf",
+    "fonts/nonscalable/sazanami-gothic.ttf", // BUG 190  crash?.  Both Scalable and Fixed Sizes
+    "fonts/drakono.ttf", // BUG 2629 ........ ??????
+    "fonts/Lazy.ttf", // BUG 2468
+    "fonts/font_bug254.ttf", // BUG 254
+    "fonts/OpenSansEmoji.ttf", // BUG 3762
+#if 1
+    "fonts/nonscalable/pvfixed_20b.pcf.gz",
+    "fonts/nonscalable/pvfixed_20r.pcf.gz",
+    "fonts/nonscalable/7x13B-ISO8859-15.pcf.gz",
+    "fonts/nonscalable/vgasys.fon", // BUG 2574
+    "fonts/nonscalable/fixed8.fon", // BUG 2127
+#endif
+#if 0
+# include "fonts/fuzzing_corpora.h"
+#endif
+#if 0
+# include "fonts/android_fonts_others.h"
+# include "fonts/android_fonts.h"
+#endif
+    "fonts/Kaumudi.ttf"
+};
+static const int test_fonts_count = sizeof(test_fonts)/sizeof(char*);
+
+
+static int wrap_size = 137;
+static int w_align = 0;
+static int outline = 0;
+static int font_style = 0;
+static int kerning = 1;
+static int wrap = 0;
+static int sdf = 0;
+static int hinting = 0;
+static int curr_str = 0;
+static int curr_font = 1;
+static int curr_size = 50;
+static int fg_alpha = 0;
+static int bg_alpha = 0;
+static int background_color = 0;
+
+static int seed = 0;
+static int print_elapsed_ticks = 0;
+static int update_screen_mode = 0;
+static int save_to_bmp = 0;
+
+/* RENDER_SOLID = 0, RENDER_BLENDED = 1, RENDER_SHADED = 2, RENDER_LCD = 3 } */
+static int render_mode = -1;
+static int render_mode_overwrite;
+static int render_mode_count = 4;
+static const char *render_mode_desc[] = { "Solid", "Blended", "Shaded", "LCD" };
+
+static int direction = 0;
+static int direction_count = 4;  
+static const char *direction_desc[] = { "LTR", "RTL", "TTB", "BTT"  };
+static const int  direction_val[] = { TTF_DIRECTION_LTR, TTF_DIRECTION_RTL, TTF_DIRECTION_TTB, TTF_DIRECTION_BTT };
```

Comment:
I would do this as:
```c
static struct {
    const char *description;
    int value;
} directions[] = {
    { "LTR", TTF_DIRECTION_LTR },
    { "RTL", TTF_DIRECTION_RTL },
    { "TTB", TTF_DIRECTION_TTB },
    { "BTT", TTF_DIRECTION_BTT }
};
```
And use `SDL_arraysize` for `direction_count`.

Reply:
yes done!
this was a test app that evolved occasionally,
there is definitively some room for improvement !